File: C++/First Steps in Coding - Basics of C++/While Loops/Exercise/Exam Preparation.cpp
#include <iostream>
#include <string>
#include <iomanip>
using namespace std;

int main() {
	int badGradesCount, badGrades=0, k=0;
	double gradesSum = 0;
	cin >> badGradesCount;
	string problemName, currProblem;
	while (true) {
		getline(cin>>ws, problemName);
		if (problemName != "Enough") {
			currProblem = problemName;
			k += 1;
			double grade;
			cin >> grade;
			gradesSum += grade;
			if (grade <= 4.00) {
				badGrades += 1;
			}
			if (badGrades == badGradesCount) {
				cout << "You need a break, " << badGrades << " poor grades.";
				break;
			}
		}
		else {
			double avgScore = (double)gradesSum / k;
			cout << fixed << setprecision(2) << "Average score: " << avgScore << "\nNumber of problems: " << k << "\nLast problem: " << currProblem;
			break;
		}
	}
}

File: Python/Basics/While Loop/Exercise/Exam Preparation.py
maxBadGrades = int(input())
limit = 0
solvedProblems = 0
lastProblem = ""
avgScore = 0

while limit < maxBadGrades:

    problemName = input()

    if problemName == "Enough":
        break
    else:
        grade = int(input())
        if grade <= 4:
            limit += 1

        solvedProblems += 1

        avgScore += grade

    lastProblem = problemName

if limit == maxBadGrades:
    print(f"You need a break, {limit} poor grades.")
else:
    print(f"Average score: {avgScore / solvedProblems:.2f}")
    print(f"Number of problems: {solvedProblems}")
    print(f"Last problem: {lastProblem}")
